### ISSUE TITLE
UnsupportedOperationException after calling addToArray multiple times

### ISSUE THREAD

**Issue by kvn-stgl** [BatchLabs/Batch-Android-SDK#19]
## Summary 

After our update to Batch 2.0.0, we receive some UnsupportedOperationException. This is probably because we call `            addToArray` mutlitple times with the same key.

The problem is probably here: https://github.com/BatchLabs/Batch-Android-SDK/blob/12359594be5f505673a01ebc52d5b90109604f43/Sources/sdk/src/main/java/com/batch/android/BatchProfileAttributeEditor.java#L369 
Array.asList will generate an immutable list but it is handled as a mutable list afterwards.

## Reproduction steps

```kotlin
            addToArray("array_partial_2", "know")
            addToArray("array_partial_2", "know_2")
```

## Logs

```
                                                                                                    Caused by: java.lang.UnsupportedOperationException
                                                                                                    	at java.util.AbstractList.add(AbstractList.java:153)
                                                                                                    	at java.util.AbstractList.add(AbstractList.java:111)
                                                                                                    	at java.util.AbstractCollection.addAll(AbstractCollection.java:336)
                                                                                                    	at com.batch.android.profile.ProfilePartialUpdateAttribute.putInAdded(ProfilePartialUpdateAttribute.java:39)
                                                                                                    	at com.batch.android.profile.ProfileUpdateOperation.addToList(ProfileUpdateOperation.java:152)
                                                                                                    	at com.batch.android.BatchProfileAttributeEditor.addToArray(BatchProfileAttributeEditor.java:369)
```

### SDK Version

2.0.1

**Comment by arnaud-roland** [comment]
Hi, thanks for the report, indeed `asList` returns a fixed-size list. We will release a patch as soon as possible.

**Comment by arnaud-roland** [comment]
Hi, this has been fixed in 2.0.2. 
Thanks again for the report.
